server/balancer.go

```diff
@@ -259,10 +271,16 @@ func (rb *resourceBalancer) doLeaderBalance(cluster *clusterInfo, stores []*stor
 	if meta.GetMaxPeerCount() == 1 {
 		addPeerOperator := newAddPeerOperator(regionID, newPeer)
 		removePeerOperator := newRemovePeerOperator(regionID, leader)
-
+		if !rb.checkScore(cluster, leader, newPeer, true) {
+			return nil, nil
+		}
 		return newBalanceOperator(region, addPeerOperator, removePeerOperator), nil
 	}
 
+	if !rb.checkScore(cluster, leader, newPeer, false) {
```

Comment:
It should be false here, the variable isLeaderPeer is `true` which means we will remove a leader peer and add a new leader peer. Otherwise it will be false.
Also you can see that only  when `meta.GetMaxPeerCount() == 1`, the variable will be set to true.